packages/quantic/cypress/e2e/default-2/generatedAnswer/generated-answer.cypress.ts

```diff
@@ -1312,6 +971,271 @@ describe('quantic-generated-answer', () => {
+          });
+
+          describe('when the generated answer is done streaming', () => {
+            it('should properly display the generated answer collapsed when the answer is too long', () => {
+              const streamId = crypto.randomUUID();
+
+              mockSearchWithGeneratedAnswer(streamId, param.useCase);
+              mockStreamResponse(streamId, genQaLongMessageTypePayload);
+              visitGeneratedAnswer({
+                collapsible: true,
+                useCase: param.useCase,
+              });
+
+              scope('when loading the page', () => {
+                Expect.displayGeneratedAnswerCard(true);
+                Expect.generatedAnswerCollapsed(true);
+                Expect.displayGeneratedAnswerToggleCollapseButton(true);
+                Expect.generatedAnswerToggleCollapseButtonContains('Show more');
+                Expect.displayCitations(false);
+                Expect.displayDisclaimer(true);
+              });
+
+              scope('when clicking the show more button to expand', () => {
+                Actions.clickToggleCollapseButton();
+                Expect.generatedAnswerCollapsed(false);
+                Expect.generatedAnswerToggleCollapseButtonContains('Show less');
+                Expect.displayCitations(false);
+                Expect.displayDisclaimer(true);
+                if (analyticsMode === 'legacy') {
+                  Expect.logGeneratedAnswerExpand(streamId);
+                }
+              });
+
+              scope(
+                'when clicking the show less button a second time to collapse',
+                () => {
+                  Actions.clickToggleCollapseButton();
+                  Expect.generatedAnswerCollapsed(true);
+                  Expect.generatedAnswerToggleCollapseButtonContains(
+                    'Show more'
+                  );
+                  Expect.displayCitations(false);
+                  Expect.displayDisclaimer(true);
+                  if (analyticsMode === 'legacy') {
+                    Expect.logGeneratedAnswerCollapse(streamId);
+                  }
+                }
+              );
+            });
+          });
+
+          it('should not display the answer collapsed when the answer is short', () => {
+            const newStreamId = crypto.randomUUID();
+
+            mockSearchWithGeneratedAnswer(newStreamId, param.useCase);
+            mockStreamResponse(newStreamId, genQaMessageTypePayload);
+            visitGeneratedAnswer({
+              collapsible: true,
+              useCase: param.useCase,
+            });
+
+            scope('when loading the page', () => {
+              Expect.displayGeneratedAnswerCard(true);
+              Expect.generatedAnswerCollapsed(false);
+              Expect.displayGeneratedAnswerToggleCollapseButton(false);
+              Expect.displayDisclaimer(true);
+            });
+          });
+        });
+
+        describe('when #withRephraseButtons is set to true', () => {
```

Comment:
The thing I think we are missing in those tests is the situations where the generated answer is collapsible,

We need to confirm that the rephrase buttons are not displayed when the answer is collapsed and that they are displayed when the answer is expanded.

Reply:
We are good with this, this would be more something we should be unit testing anyway. (especially since our e2e simply mock the responses).

The condition is litteraly `!this.isAnswerCollapsed` so I don't feel like we need an e2e test to make sure that when the answer is collapsed, the rephrase buttons are not showing :) 

Also we are planning on removing them soon.